crates/joystick/src/linux.rs

```diff
@@ -298,72 +96,111 @@ impl Default for Joystick {
             .unwrap();
 
         Joystick {
-            events: vec![],
             device: Arc::new(Mutex::new(device)),
-            instant: Instant::now(),
             keys,
         }
     }
 }
 
+// To visually run these tests, use:
+// `cargo test -- --test-threads 1`
+// then focus https://hardwaretester.com/gamepad 
 #[cfg(test)]
 mod tests {
-    use crate::common::JoystickInterface;
+    use crate::common::{JoystickInterface, Button};
     use crate::joystick::Joystick;
     use std::thread::sleep;
-    use std::time::{Duration, Instant};
+    use std::time::Duration;
 
     #[test]
     fn test_event_system() -> std::io::Result<()> {
-        sleep(Duration::from_millis(1000));
+        sleep(Duration::from_millis(2000));
         let mut joystick: Joystick = Joystick::default();
-        joystick.tap_dpad_up();
         sleep(Duration::from_millis(500));
-        joystick.tap_dpad_down();
+        joystick.press(Button::UP);
```

Comment:
Should be noted that these no longer do the same thing, since `press` and `tap` are different.